admin/class-sitemaps-admin.php

```diff
@@ -59,7 +68,13 @@ function delete_sitemaps() {
 	 * @param string   $old_status Old post status.
 	 * @param \WP_Post $post       Post object.
 	 */
-	function status_transition( $new_status, $old_status, $post ) {
+	public function status_transition( $new_status, $old_status, $post ) {
+		if ( defined( 'WP_IMPORTING' ) ) {
+			$this->status_transition_bulk( $new_status, $old_status, $post );
+
+			return;
+		}
```

Comment:
Thanks for including the `WP_IMPORTING` check! — Now, I'm not sure whether registering each post for a bulk execution is the proper thing to do.

#3879 simply returned without doing anything else for imported posts, because my train of thought was:

1. An imported post most likely existed on the site before already - so the proper point in time to notify search engines was when the post was originally published, but not now, because the post is just "moving".  Would you agree?

2. It looks like this new code still attempts to notify search engines for every single post that has been imported; the process merely happens at the very end of an import operation (when the `'admin_footer'` hook is fired) - which doesn't seem to improve or resolve the original performance issue...  I'm importing 800.000 posts, which means 800.000 pings to search engines...

3. I never executed a data import or migration on a production site in my entire life, so I would bet that in most cases, all of these pings would try to inform search engines about newly imported posts on staging sites or private/local development sites.

That's why I muted the entire operation during imports.  Thoughts?